### PR TITLE
initial Twitter oauth support via passport

### PR DESCRIPTION
**Pull request by mderazon**
botim/backend#1

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,2 +0,0 @@
-PORT=8080
```
Comment (eranshmil) [inline .env.example:1]: Why did you remove this file?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (eranshmil) [inline .prettierrc:1]: I prefer this configuration, just like in the extension. :)
```json 
{
  "singleQuote": true,
  "printWidth": 95,
  "tabWidth": 2
}
```

File: src/app.js [hunk 3]
```diff
@@ -1,40 +1,36 @@
...
 
 const app = express();
-app.disable('x-powered-by');
 
-// View engine setup
-app.set('views', path.join(__dirname, '../views'));
-app.set('view engine', 'pug');
+app.use(helmet());
+app.use(morgan('combined'));
+app.use(cookieParser());
```
Comment (eranshmil) [inline src/app.js:14]: The `cookieParser` and the `expressSession` are required for the twitter oauth?

File: .prettierrc (hunk 2 above)
Comment (mderazon) [reply .prettierrc:1]: np, changed

File: .env.example (hunk 1 above)
Comment (mderazon) [reply .env.example:1]: I just moved it to [.env](https://github.com/botim/backend/blob/869d1193bf8eb04790b74081a629dc608407f40b/.env) file. gitignore had `.env` so it didn't commit it so I committed it manually for reference for now but I think it's good to keep it in gitignore so we don't accidentally commit secrets

File: src/app.js (hunk 3 above)
Comment (mderazon) [reply src/app.js:14]: yes, express session is used to create a session cookie and cookie parser to parse that cookie.

**Comment by mderazon** [comment]
These are the initial routes needed for authentication. It uses passportjs twiter auth strategy (http://www.passportjs.org/docs/twitter/). 

Once authenticated, it uses a cookie to store the session so that next time the client (extension) will just have to send the cookie.
Another point is that we do not use our own user model and we don't store the user in the database, all we do is store the user details we got from Twiter in the cookie so we can decode them on every request.
We need to think if we need to store the user in the db at all, for now I didn't see a use case for it and less = better.

I am still waiting to test this, I could not get keys from Twitter as they haven't approved my request for dev account yet and it doesn't seem like they have a sandbox environment for logging in for non dev accounts

**Comment by eranshmil** [comment]
This feature is not needed anymore.
Thanks for your contribution and help!
